build/lib/src/asset/id.dart

```diff
@@ -19,6 +19,9 @@ class AssetId implements Comparable<AssetId> {
   /// the host platform.
   final String path;
 
+  /// Splits [path] into its components.
+  List<String> get pathSegments => p.posix.split(path);
```

Comment:
Generally we use `p.url`, I don't know where that is different but it would be good to be consistent.